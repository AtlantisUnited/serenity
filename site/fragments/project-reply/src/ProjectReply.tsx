import React                        from 'react'
import { WithUser, WithoutUser }    from '@atls/react-user'

import ChooseSpecialistModal        from '@fragments/choose-specialist-modal'
import { Avatar }                   from '@ui/avatar'
import { Button }                   from '@ui/button'
import { Divider }                  from '@ui/divider'
import { StarIcon, ThumbUpIcon }    from '@ui/icons'
import { Input, Textarea }          from '@ui/input'
import { Box, Column, Layout, Row } from '@ui/layout'
import { Link, SidebarLink }        from '@ui/link'
import { Option, Select }           from '@ui/select'
import { Text }                     from '@ui/text'

import messages                     from './messages'

export const ProjectReply = ({
  intl,
  message,
  profile = {},
  replies,
  loading,
  errors,
  comments,
  visible,
  status,
  ownerName,
  replyChooseData,
  activeFilter,
  onChangeMessage,
  onChangeComment,
  onSaveMessage,
  onRegistration,
  onLogin,
  onSaveComment,
  onChangeStatus,
  onChooseSpecialist,
  closeChooseSpecialist,
  onOpenSpecialist,
  onAcceptProject,
  onRejectProject,
  onFilterReplies,
}: any) => (
  <>
    <WithUser>
      {profile && (profile.type === 'specialist' || profile.type === 'owner') && (
        <Box bg='alabaster' alignItems='center' flexDirection='column'>
          <Divider />
          <Layout flexBasis={25} />
          <Box
            maxWidth={1200}
            px={[20, 20, 40]}
            boxSizing='border-box'
            width='100%'
            flexDirection={['column-reverse', 'row']}
          >
            <Layout width='100%'>
              <Column>
                <Layout flexBasis={48} alignItems='baseline'>
                  <Text fontSize='giant' lineHeight='medium'>
                    {intl.formatMessage(
                      profile.type === 'owner' ? messages.responses : messages.yourResponse
                    )}
                  </Text>
                  {profile.type === 'owner' && (
                    <>
                      <Layout flexBasis={32} />
                      <Text fontSize='huge' color='silver' fontWeight='medium'>
                        {replies.length}
                      </Text>
                    </>
                  )}
                </Layout>
                {profile.type === 'specialist' && replies[0] && replies[0].status === 'chosen' && (
                  <Box
                    mt={24}
                    py={24}
                    width='100%'
                    border='black'
                    borderRadius='extra'
                    flexDirection='column'
                    overflow='hidden'
                    bg='wildsand'
                  >
                    <Row>
                      <Layout flexBasis={52} />
                      <Layout>
                        <ThumbUpIcon width={20} height={20} />
                      </Layout>
                      <Layout flexBasis={16} flexShrink={0} />
                      <Layout flexGrow={1}>
                        <Column>
                          <Layout>
                            <Text lineHeight='medium'>
                              <Text fontWeight='bold'>
                                {`${ownerName.firstName} ${ownerName.lastName}`}
                              </Text>
                              {intl.formatMessage(messages.projectOffer)}
                            </Text>
                          </Layout>
                          <Layout flexBasis={26} />
                          <Layout>
                            <Button borderRadius='large' onClick={onRejectProject}>
                              {intl.formatMessage(messages.projectReject)}
                            </Button>
                            <Layout flexBasis={16} flexShrink={0} />
                            <Button color='chicago' borderRadius='large' onClick={onAcceptProject}>
                              {intl.formatMessage(messages.projectAccept)}
                            </Button>
                          </Layout>
                        </Column>
                      </Layout>
                      <Layout flexBasis={24} flexShrink={0} />
                    </Row>
                  </Box>
                )}
                <Layout flexBasis={28} />
                {replies.length === 0 && profile.type !== 'owner' && (
                  <Layout>
                    <Row>
                      <Layout>
                        <Avatar
                          src={profile.photo && profile.photo.url}
                          width={56}
                          height={56}
                          color='alto'
                        />
                      </Layout>
                      <Layout flexBasis={22} />
                      <Layout flexGrow={1}>
                        <Column>
                          {errors.message && (
                            <Layout pb='4px'>
                              <Text color='red'>{errors.message}</Text>
                            </Layout>
                          )}
                          <Layout>
                            <Textarea
                              boxShadow='black'
                              rows={6}
                              value={message}
                              onChange={onChangeMessage}
                              placeholder={intl.formatMessage(messages.messagePlaceholder)}
                            />
                          </Layout>
                          <Layout flexBasis={22} />
                          <Layout>
                            <Button onClick={onSaveMessage} disabled={loading} color='chicago'>
                              {intl.formatMessage(messages.respondButton)}
                            </Button>
                          </Layout>
                        </Column>
                      </Layout>
                    </Row>
                  </Layout>
                )}
                {replies.length > 0 &&
                  replies.map((reply) => (
                    <Box
                      key={reply.id}
                      my='7px'
                      width='100%'
                      border='black'
                      borderRadius='extra'
                      flexDirection='column'
                      overflow='hidden'
                      bg='white'
                    >
                      {reply.discussion.messages.map((replyMessage, index) => (
                        <Column key={replyMessage.id}>
                          <Layout flexBasis={index === 0 ? 16 : 24} />
                          <Layout pr={['16px', '24px']} pl={16}>
                            <Row>
                              <Layout>
                                <Column>
                                  <Layout
                                    width={56}
                                    height={56}
                                    alignItems='center'
                                    justifyContent='flex-end'
                                  >
                                    <Avatar
                                      width={index === 0 ? 56 : 32}
                                      height={index === 0 ? 56 : 32}
                                      color='alto'
                                      src={
                                        (replyMessage.author.photo &&
                                          replyMessage.author.photo.url) ||
                                        null
                                      }
                                    />
                                  </Layout>
                                  <Layout flexBasis={13} />
                                  {index === 0 && (
                                    <Layout alignItems='center' justifyContent='center'>
                                      <Layout>
                                        <Text lineHeight='medium'>
                                          {(replyMessage.member && replyMessage.member.rating) ||
                                            0.0}
                                        </Text>
                                      </Layout>
                                      <Layout pl='4px'>
                                        <StarIcon width={16} height={16} />
                                      </Layout>
                                    </Layout>
                                  )}
                                </Column>
                              </Layout>
                              <Layout flexBasis={16} flexShrink={0} />
                              <Layout flexGrow={1} overflow='hidden'>
                                <Column>
                                  <Layout flexBasis='7px' />
                                  <Layout>
                                    <Row justifyContent='space-between' alignItems='flex-start'>
                                      <Layout>
                                        <Column>
                                          <Layout>
                                            <Text
                                              fontWeight='medium'
                                              fontSize='regular'
                                              lineHeight='medium'
                                              color='black'
                                              as={replyMessage.member ? Link : Text}
                                              onClick={() => {
                                                if (replyMessage.member) {
                                                  onOpenSpecialist(replyMessage.member.id)
                                                }
                                              }}
                                            >
                                              {replyMessage.member &&
                                              replyMessage.member.interaction.formOfWork ===
                                                'company'
                                                ? replyMessage.member.interaction.name
                                                : `${replyMessage.author.personalInformation.firstName} ${replyMessage.author.personalInformation.lastName}`}
                                            </Text>
                                          </Layout>
                                          {index === 0 &&
                                            replyMessage.member &&
                                            replyMessage.member.specialisation.main && (
                                              <Layout alignItems='center'>
                                                <Layout>
                                                  <Text
                                                    color='silver'
                                                    lineHeight='medium'
                                                    wordBreak='break-word'
                                                  >
                                                    {replyMessage.member.specialisation.main
                                                      .map((item) => item.name)
                                                      .join(', ')}
                                                  </Text>
                                                </Layout>
                                                <Layout>
                                                  {replyMessage.member.specialisation.additional &&
                                                    replyMessage.member.specialisation
                                                      .additional[0] && (
                                                      <Box
                                                        bg='alto'
                                                        ml='12px'
                                                        px='4px'
                                                        height={20}
                                                        alignItems='center'
                                                        display='inline-block'
                                                      >
                                                        <Text
                                                          color='silver'
                                                          fontWeight='medium'
                                                        >{`+${replyMessage.member.specialisation.additional.length}`}</Text>
                                                      </Box>
                                                    )}
                                                </Layout>
                                              </Layout>
                                            )}
                                        </Column>
                                      </Layout>
                                      <Layout flexBasis={20} flexShrink={0} />
                                      {profile.type === 'owner' && index === 0 && (
                                        <Layout flexBasis={144}>
                                          <Select
                                            rounded
                                            disabled={
                                              status === 'selected' ||
                                              status === 'performed' ||
                                              status === 'completed'
                                            }
                                            value={reply.status}
                                            onChange={(value) =>
                                              onChangeStatus(value, status, {
                                                photo: replyMessage.author.photo,
                                                personalInformation:
                                                  replyMessage.author.personalInformation,
                                                id: reply.id,
                                              })
                                            }
                                            optionLabelProp='label'
                                          >
                                            <Option
                                              className='hidden'
                                              key='new'
                                              value='new'
                                              label={intl.formatMessage(messages.newRequest)}
                                            >
                                              {intl.formatMessage(messages.newRequest)}
                                            </Option>
                                            <Option
                                              className='hidden'
                                              key='rejected'
                                              value='rejected'
                                              label={intl.formatMessage(messages.denied)}
                                            >
                                              {intl.formatMessage(messages.denied)}
                                            </Option>
                                            <Option
                                              key='candidate'
                                              value='candidate'
                                              label={intl.formatMessage(messages.candidate)}
                                            >
                                              {intl.formatMessage(messages.candidate)}
                                            </Option>
                                            <Option
                                              key='chosen'
                                              value='chosen'
                                              label={intl.formatMessage(messages.chosen)}
                                            >
                                              {intl.formatMessage(messages.choose)}
                                            </Option>
                                            <Option
                                              key='denied'
                                              value='denied'
                                              label={intl.formatMessage(messages.denied)}
                                            >
                                              {intl.formatMessage(messages.deny)}
                                            </Option>
                                          </Select>
                                        </Layout>
                                      )}
                                    </Row>
                                  </Layout>
                                  <Layout flexBasis={18} />
                                  <Layout>
                                    <Text lineHeight='medium'>{replyMessage.content}</Text>
                                  </Layout>
                                </Column>
                              </Layout>
                            </Row>
                          </Layout>
                          <Layout flexBasis={24} />
                          {index + 1 < reply.discussion.messages.length && <Divider />}
                        </Column>
                      ))}
                      <Divider />
                      <Layout flexBasis={24} />
                      {errors.message && (
                        <Layout pb='4px'>
                          <Text color='red'>{errors.message}</Text>
                        </Layout>
                      )}
                      <Layout px={24}>
                        <Input
                          size='medium'
                          prefix={
                            <Avatar
                              width={32}
                              height={32}
                              color='alto'
                              src={(profile.photo && profile.photo.url) || null}
                            />
                          }
                          value={comments[reply.id]}
                          placeholder={intl.formatMessage(messages.commentPlaceholder)}
                          onChange={(value) => onChangeComment(value, reply.id)}
                          onEnter={() => onSaveComment(reply.id)}
                        />
                      </Layout>
                      <Layout flexBasis={24} />
                    </Box>
                  ))}
              </Column>
            </Layout>
            <Layout flexBasis={24} flexShrink={0} />
            <Layout pt={[20, 83]} flexBasis={['auto', 206, 256]} flexShrink={0}>
              {profile.type === 'owner' && (
                <Column>
                  <Box border='black' borderRadius='extra' boxSizing='border-box' bg='white'>
                    <Column>
                      <Layout>
                        <SidebarLink
                          active={activeFilter === ''}
                          onClick={() => onFilterReplies('')}
                        >
                          {intl.formatMessage(messages.allResponsesSidebar)}
                        </SidebarLink>
                      </Layout>
                      <Divider />
                      <Layout>
                        <SidebarLink
                          active={activeFilter === 'new'}
                          onClick={() => onFilterReplies('new')}
                        >
                          {intl.formatMessage(messages.newResponsesSidebar)}
                        </SidebarLink>
                      </Layout>
                      <Divider />
                      <Layout>
                        <SidebarLink
                          active={activeFilter === 'candidate'}
                          onClick={() => onFilterReplies('candidate')}
                        >
                          {intl.formatMessage(messages.candidatesSidebar)}
                        </SidebarLink>
                      </Layout>
                      <Divider />
                      <Layout>
                        <SidebarLink
                          active={activeFilter === 'denied'}
                          onClick={() => onFilterReplies('denied')}
                        >
                          {intl.formatMessage(messages.deniedSidebar)}
                        </SidebarLink>
                      </Layout>
                    </Column>
                  </Box>
                </Column>
              )}
            </Layout>
          </Box>
          <Layout flexBasis={[40, 60, 80]} />
        </Box>
      )}
      <ChooseSpecialistModal
        visible={visible}
        data={replyChooseData}
        onClose={closeChooseSpecialist}
        onSave={onChooseSpecialist}
      />
    </WithUser>
    <WithoutUser>
      <Divider />
      <Box justifyContent='center' bg='alabaster'>
        <Row maxWidth={1200} px={[20, 20, 40]} boxSizing='border-box' width='100%'>
          <Layout>
            <Column>
              <Layout flexBasis={25} />
              <Layout>
                <Text fontSize='giant' lineHeight='medium'>
                  {intl.formatMessage(messages.response)}
                </Text>
              </Layout>
              <Layout flexBasis={28} />
              <Box p={24} borderRadius='4px' bg='wildsand' border='black' flexDirection='column'>
                <Layout>
                  <Text lineHeight='medium'>{intl.formatMessage(messages.registration)}</Text>
                </Layout>
                <Layout flexBasis={27} />
                <Layout>
                  <Box width='100%' flexDirection={['column', 'row']}>
                    <Layout>
                      <Button onClick={onRegistration} borderRadius='large' color='chicago' fill>
                        {intl.formatMessage(messages.registrationButton)}
                      </Button>
                    </Layout>
                    <Layout flexBasis={16} />
                    <Layout>
                      <Button onClick={onLogin} borderRadius='large' fill>
                        <Layout px={14}>{intl.formatMessage(messages.loginButton)}</Layout>
                      </Button>
                    </Layout>
                  </Box>
                </Layout>
              </Box>
              <Layout flexBasis={[40, 60, 80]} />
            </Column>
          </Layout>
          <Layout flexBasis={['auto', 220, 296]} flexShrink={0} />
        </Row>
      </Box>
    </WithoutUser>
  </>
)
