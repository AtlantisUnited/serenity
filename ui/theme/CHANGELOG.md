# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.5 (2021-05-10)

**Note:** Version bump only for package @ui/theme

## 0.0.4 (2020-09-28)

### Bug Fixes

- **accounts:** add emotion-server, delete font ([08acf5a](https://github.com/Atlantis-Lab/serenity/commit/08acf5a7169352a606925fdfbbaa18f9917a8106))

## 0.0.3 (2020-06-30)

**Note:** Version bump only for package @ui/theme

## 0.0.2 (2020-06-10)

**Note:** Version bump only for package @ui/theme
