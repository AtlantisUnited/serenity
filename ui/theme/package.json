{
  "name": "@ui/theme",
  "version": "0.0.5",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@types/react": "^17.0.6",
    "@types/styled-system": "5.1.4"
  },
  "dependencies": {
    "@emotion/css": "^11.1.3",
    "@emotion/react": "^11.4.0",
    "@emotion/styled": "^11.3.0",
    "react": "^17.0.2",
    "styled-system": "^5.1.0"
  }
}
