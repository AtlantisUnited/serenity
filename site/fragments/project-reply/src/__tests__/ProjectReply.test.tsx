import React                     from 'react'
import { IntlProvider, useIntl } from 'react-intl'

import { MockedProvider }        from '@apollo/react-testing'
import { ThemeProvider }         from '@ui/theme'
import { render }                from '@testing-library/react'

import { ProjectReply }          from '../ProjectReply'

const TestComponent = () => {
  const intl = useIntl()

  return <ProjectReply intl={intl} />
}

describe('Project reply', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <MockedProvider mocks={[]} addTypename={false}>
          <IntlProvider locale='ru' defaultLocale='ru' messages={{}}>
            <ThemeProvider>
              <TestComponent />
            </ThemeProvider>
          </IntlProvider>
        </MockedProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
