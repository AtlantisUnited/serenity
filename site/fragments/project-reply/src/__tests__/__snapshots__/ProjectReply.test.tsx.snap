// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project reply snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  height: 1px;
  background: rgb(235,235,235);
}

<div
    class="emotion-0"
  />
  .emotion-21 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: rgb(252,252,252);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-20 {
  max-width: 1200px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 20px;
  padding-right: 20px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
}

@media screen and (min-width:40em) {
  .emotion-20 {
    padding-left: 20px;
    padding-right: 20px;
  }
}

@media screen and (min-width:52em) {
  .emotion-20 {
    padding-left: 40px;
    padding-right: 40px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-17 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 25px;
  -ms-flex-preferred-size: 25px;
  flex-basis: 25px;
}

.emotion-1 {
  font-size: 28px;
  line-height: 1.6;
  font-family: Inter;
  font-weight: 400;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 28px;
  -ms-flex-preferred-size: 28px;
  flex-basis: 28px;
}

.emotion-15 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 24px;
  background-color: rgb(250,250,250);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,0.08);
}

.emotion-4 {
  line-height: 1.6;
  font-family: Inter;
  font-weight: 400;
  font-size: 13px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 27px;
  -ms-flex-preferred-size: 27px;
  flex-basis: 27px;
}

.emotion-13 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media screen and (min-width:40em) {
  .emotion-13 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-7 {
  font-family: Inter;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  cursor: pointer;
  border: none;
  outline: none;
  padding: 0;
  width: 100%;
  font-size: 12px;
  padding: 0px 16px;
  height: 40px;
  background: rgb(87,87,87);
  color: rgb(255,255,255);
  border-radius: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-7:active {
  -webkit-transform: scale(0.98);
  -ms-transform: scale(0.98);
  transform: scale(0.98);
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 16px;
  -ms-flex-preferred-size: 16px;
  flex-basis: 16px;
}

.emotion-11 {
  font-family: Inter;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  cursor: pointer;
  border: none;
  outline: none;
  padding: 0;
  width: 100%;
  font-size: 12px;
  padding: 0px 16px;
  height: 40px;
  border: 1px solid rgba(0,0,0,0.08);
  color: rgb(158,158,158);
  background: rgb(255,255,255);
  border-radius: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-11:active {
  -webkit-transform: scale(0.98);
  -ms-transform: scale(0.98);
  transform: scale(0.98);
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 14px;
  padding-right: 14px;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 40px;
  -ms-flex-preferred-size: 40px;
  flex-basis: 40px;
}

@media screen and (min-width:40em) {
  .emotion-16 {
    -webkit-flex-basis: 60px;
    -ms-flex-preferred-size: 60px;
    flex-basis: 60px;
  }
}

@media screen and (min-width:52em) {
  .emotion-16 {
    -webkit-flex-basis: 80px;
    -ms-flex-preferred-size: 80px;
    flex-basis: 80px;
  }
}

.emotion-19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media screen and (min-width:40em) {
  .emotion-19 {
    -webkit-flex-basis: 220px;
    -ms-flex-preferred-size: 220px;
    flex-basis: 220px;
  }
}

@media screen and (min-width:52em) {
  .emotion-19 {
    -webkit-flex-basis: 296px;
    -ms-flex-preferred-size: 296px;
    flex-basis: 296px;
  }
}

<div
    class="emotion-21"
    display="flex"
  >
    <div
      class="emotion-20"
      display="flex"
      width="100%"
    >
      <div
        class="emotion-2"
        display="flex"
      >
        <div
          class="emotion-17"
          display="flex"
          width="100%"
        >
          <div
            class="emotion-0"
            display="flex"
          />
          <div
            class="emotion-2"
            display="flex"
          >
            <span
              class="emotion-1"
              font-family="primary"
              font-size="giant"
              font-weight="normal"
            >
              Отклик
            </span>
          </div>
          <div
            class="emotion-3"
            display="flex"
          />
          <div
            class="emotion-15"
            display="flex"
          >
            <div
              class="emotion-2"
              display="flex"
            >
              <span
                class="emotion-4"
                font-family="primary"
                font-size="normal"
                font-weight="normal"
              >
                Чтобы откликнуться на этот проект, вам необходимо зарегистрироваться на сервисе как исполнитель. Вы можете сделать это прямо сейчас:
              </span>
            </div>
            <div
              class="emotion-6"
              display="flex"
            />
            <div
              class="emotion-2"
              display="flex"
            >
              <div
                class="emotion-13"
                display="flex"
                width="100%"
              >
                <div
                  class="emotion-2"
                  display="flex"
                >
                  <button
                    class="emotion-7"
                  >
                    Зарегистрировать аккаунт
                  </button>
                </div>
                <div
                  class="emotion-9"
                  display="flex"
                />
                <div
                  class="emotion-2"
                  display="flex"
                >
                  <button
                    class="emotion-11"
                  >
                    <div
                      class="emotion-10"
                      display="flex"
                    >
                      Войти
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="emotion-16"
            display="flex"
          />
        </div>
      </div>
      <div
        class="emotion-19"
        display="flex"
      />
    </div>
  </div>
</DocumentFragment>
`;
